### PR TITLE
Added Assign Public IP on Launch

### PR DESCRIPTION
**Pull request by nadenf**
## what
Provides option to auto-assign public IP addresses for public subnets.

## why
EKS needs auto-assign public IP enabled for public subnets.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -117,3 +117,7 @@ variable "nat_gateway_enabled" {
   default     = "true"
 }
 
+variable "assign_public_ip_on_launch" {
+  description = "Flag to enable/disable whether instances launched into public subnets should be assigned a public IP address"
+  default     = "false"
```
Comment (Gowiem) [inline variables.tf:122]: @nadenf please update to use `type = bool` and update the default to use the bool `false` instead of the string `"false"`?

**Review by Gowiem — COMMENTED** [review]
@nadenf One request -- Please update and we'll get this reviewed and merged.

**Comment by ruiengana** [comment]
Is this PR planned to be merged?

**Comment by nathanielvarona** [comment]
Any plans on clossing or merging this PR?

**Pull request closed without merging**